spyder/plugins/projects/plugin.py

```diff
@@ -212,30 +217,45 @@ def on_completions_available(self):
         #         self.start_workspace_services())
         self.completions.sig_stop_completions.connect(
             self.stop_workspace_services)
-        self.sig_project_loaded.connect(
-            functools.partial(self.completions.project_path_update,
-                              update_kind=WorkspaceUpdateKind.ADDITION,
-                              instance=self))
-        self.sig_project_closed.connect(
-            functools.partial(self.completions.project_path_update,
-                              update_kind=WorkspaceUpdateKind.DELETION,
-                              instance=self))
+
+        self._addition_path_update = functools.partial(
+            self.completions.project_path_update,
+            update_kind=WorkspaceUpdateKind.ADDITION,
+            instance=self)
+
+        self._deletion_path_update = functools.partial(
+            self.completions.project_path_update,
+            update_kind=WorkspaceUpdateKind.DELETION,
+            instance=self)
+
+        self.sig_project_loaded.connect(self._addition_path_update)
+        self.sig_project_closed.connect(self._deletion_path_update)
 
     @on_plugin_available(plugin=Plugins.IPythonConsole)
     def on_ipython_console_available(self):
         self.ipyconsole = self.get_plugin(Plugins.IPythonConsole)
         widget = self.get_widget()
         treewidget = widget.treewidget
 
-        treewidget.sig_open_interpreter_requested.connect(
-            self.ipyconsole.create_client_from_path)
-        treewidget.sig_run_requested.connect(
+        self._ipython_run_script = (
             lambda fname:
-            self.ipyconsole.run_script(
-                fname, osp.dirname(fname), '', False, False, False, True,
-                False)
+                self.ipyconsole.run_script(
+                    fname, osp.dirname(fname), '', False, False, False, True,
+                    False
+                )
         )
 
+        treewidget.sig_open_interpreter_requested.connect(
+            self.ipyconsole.create_client_from_path)
+        treewidget.sig_run_requested.connect(self._ipython_run_script)
+
+    @on_plugin_available(plugin=Plugins.OutlineExplorer)
+    def on_outline_explorer_available(self):
+        outline_explorer = self.get_plugin(Plugins.OutlineExplorer)
+        self._outline_update = lambda v: outline_explorer.update_all_editors()
+        self.sig_project_loaded.connect(self._outline_update)
+        self.sig_project_closed.connect(self._outline_update)

```

Comment:
This is not necessary anymore because I removed the connection between Projects and the Outline explorer.